hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -568,6 +561,39 @@ private void checkIterationMoveResults() {
         metrics.getNumContainerMovesCompletedInLatestIteration());
   }
 
+  private long performTimeMoveCancelAndCount() {
+    return moveSelectionToFutureMap.entrySet().stream()
+        .filter(entry -> !entry.getValue().isDone())
+        .peek(entry -> {
+          LOG.warn("Container move canceled for container {} from source {}" +
+                  " to target {} due to timeout.",
+              entry.getKey().getContainerID(),
+              containerToSourceMap.get(entry.getKey().getContainerID())
+                  .getUuidString(),
+              entry.getKey().getTargetNode().getUuidString());
+          entry.getValue().cancel(true);
+        }).count();
+  }
+
+  private long performMoveCancelAndFailCount() {
+    long cnt = moveSelectionToFutureMap.entrySet().stream()
+        .filter(entry -> entry.getValue().isDone() && entry.getValue()
+            .isCompletedExceptionally()).count();
+    moveSelectionToFutureMap.entrySet().stream()
+        .filter(entry -> !entry.getValue().isDone())
+        .peek(entry -> {
+          LOG.warn("Container move canceled for container {} from source {}" +
+                  " to target {} due to timeout.",
```

Comment:
Should be "failure" instead of "timeout".

Reply:
removed the specific method as completeExceptionally also not applicable for interrup exception and will be counted in whenComplete(), reused cancelAndCountPendingMoves()